actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -283,6 +283,8 @@ def parse(name, signed_message)
         end
     end
 
+    mattr_accessor :cookies_have_purpose, default: false
```

Comment:
We don't need a config for this, we just need to support cookies with and without a purpose. In other words let new cookies be written with a purpose but still be able to read the old ones out too (backwardscompatibility).